spec/enumerable_spec.rb

```diff
@@ -0,0 +1,102 @@
+require_relative('../main')
+describe Enumerable do
+  describe '#my_each' do
+    it 'returns enumerator' do
+      expect(%i[1 2 4 5 6].my_each).to(be_an(Enumerator))
+    end
+
+    it 'my_each should return the same as each' do
+      expect(%i[1 2 4 5 6].my_each.to_a).to(eql(%i[1 2 4 5 6].each.to_a))
+    end
+
+    it 'should returns key, value pair' do
+      expect({ name: 'jay', age: 34 }.my_each.to_h).to(eq({ name: 'jay', age: 34 }.each.to_h))
+    end
+
+    it 'returns elements in the range' do
+      expect(%i[0..10].my_each.to_a).to(eq(%i[0..10].each.to_a))
+    end
+  end
```

Comment:
I can see a lot of repetitive arrays, hash, range. Which is not a good sign of clean code. First of all, you can assign them in a variable. then you can them whenever needed. In Rspec we assign variables using `let`.And my_each is not checking for some edge case. I tried my best to organize your test in this format. You can either replace your one with it or you can generate a new case using the following format.
```ruby
describe '#my_each' do
    let(:array) { [1, 2, 4, 5, 6] }
    let(:range) {Range.new(0, 5)}
    let(:hash) { {name: 'jay', age: 34} }
    it 'returns enumerator' do
      expect(array.my_each).to be_an(Enumerator)
    end

    it 'calls the block for each of the item in the array' do
      each_output = 0
      block = proc { |num| each_output += num }
      array.my_each(&block)
      my_each_output = each_output.dup
      each_output = 0
      array.each(&block)
      expect(my_each_output).to eql(each_output)
    end

    it 'calls the block for each of the item in the hash' do
      each_output = ''
      block = proc { |key, val| each_output += "#{key} #{val}" }
      hash.my_each(&block)
      my_each_output = each_output.dup
      each_output = ''
      hash.each(&block)
      expect(my_each_output).to eql(each_output)
    end

    it 'calls the block for each of the item in the array' do
      each_output = 0
      block = proc { |num| each_output += num }
      range.my_each(&block)
      my_each_output = each_output.dup
      each_output = 0
      range.each(&block)
      expect(my_each_output).to eql(each_output)
    end
end
```